torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2729,8 +2729,12 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collective(
   at::cuda::OptionalCUDAGuard gpuGuard(device);
 
   if (nanCheck) {
+    at::cuda::CUDAStreamGuard ncclStreamGuard(ncclStream);
     for (const auto& input : inputs) {
-      checkForNan(input, ncclStream);
+      bool nan = isnan(input)._is_any_true().item<bool>();
+      if (nan) {
+        throw std::runtime_error("NaN check failed in collective()");
+      }
```

Comment:
In case we are supporting both options, maybe we can remove the `ncclStream` signature from `checkForNan` and share the same stream guard here?